consumer.go

```diff
@@ -104,6 +105,23 @@ func (c *Consumer) Scan(ctx context.Context, fn ScanFunc) error {
 		shardc = make(chan *kinesis.Shard, 1)
 	)
 
+	// Discover the retention period on the stream, which is used later in ScanShard()
+	// for failure mode handling.
+	summary, err := c.client.DescribeStreamSummaryWithContext(
+		aws.Context(ctx),
+		&kinesis.DescribeStreamSummaryInput{
+			StreamName: aws.String(c.streamName),
+		},
+	)
+	if err != nil {
+		if err.(awserr.Error).Code() != "AccessDeniedException"  {
+			return err
+		}
+		c.logger.Log("[CONSUMER] IAM entity lacks kinesis:DescribeStreamSummary permissions, skipping extra sanity checks")
```

Comment:
Existing users may have IAM policies crafted to work with kinesis-consumer that don't allow this permission, so it's optional.